tests/resource/test_error.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Test base AlgorithmicError class and its associated methods
+"""
+import pytest
+
+import pennylane as qml
+from pennylane.resource.error import AlgorithmicError
+
+
+class SimpleError(AlgorithmicError):
+    def combine(self, other):
+        return self.__class__(self.error + other.error)
+
+    @staticmethod
+    def get_error(approx_op, other_op):  # pylint: disable=unused-argument
+        return 0.5  # get simple error is always 0.5
+
+
+class ErrorNoCombine(AlgorithmicError):  # pylint: disable=too-few-public-methods
+    @staticmethod
+    def get_error(approx_op, other_op):  # pylint: disable=unused-argument
+        return 0.5
+
+
+class ErrorNoGetError(AlgorithmicError):  # pylint: disable=too-few-public-methods
+    def combine(self, other):
+        return self.__class__(self.error + other.error)
+
+
+class TestAlgorithmicError:
+    """Test the methods and attributes of the Resource class"""
+
+    @pytest.mark.parametrize("error", [1.23, 0.45, -6])
+    def test_error_attribute(self, error):
+        """Test that instantiation works"""
+        ErrorObj = SimpleError(error)
+        assert ErrorObj.error == error
+
+    def test_combine_not_implemented(self):
+        """Test NotImplementedError is raised if the method is not defined."""
+        ErrorObj1 = ErrorNoCombine(1.23)
+        ErrorObj2 = ErrorNoCombine(0.45)
```

Comment:
Do we expect `NotImplementedError` or failure in instantiating the class? What is the reason for making combine  `@abstractmethod`?

Reply:
This was a result of forgetting to inherit from ABC. I added it and fixed the test. We want it to fail on instantiation, not on call of the method as in the old test.